src/main/storm/mesos/MesosNimbus.java

```diff
@@ -789,15 +789,11 @@ protected void computeResourcesForSlot(final RotatingMap<OfferID, Offer> offers,
         executorInfoBuilder
             .setName(executorName)
             .setExecutorId(ExecutorID.newBuilder().setValue(details.getId()))
-            .setData(ByteString.copyFromUtf8(executorDataStr));
-        if (!subtractedExecutorResources) {
-          subtractedExecutorResources = true;
```

Comment:
Soo... my fix is buggy.  The 2nd time through the loop where `subtractedExecutorResources` is `true`, it won't enter the block that sets the `executorCpuResources` and `executorMemResources` variables, which causes an NPE further down when building the `ExecutorInfo`.

The immediate fix is simple, just set those variables unconditionally... but that led me to read [the logviewer port-acquisition logic](https://github.com/mesos/storm/blob/7333f9e02a13a747c50f53992831630f7cd1eea6/src/main/storm/mesos/MesosNimbus.java#L759-L770).  I'm pretty sure it has multiple issues.  Gotta test more...